src/util/isIterable.ts

```diff
@@ -0,0 +1,5 @@
+import { iterator as Symbol_iterator } from '../symbol/iterator';
+
+export function isIterable<T>(value: any): value is Iterable<T> {
```

Comment:
It helps flow the types through, `isPromise` and `isArrayLike` are also both generic.